### ISSUE TITLE
Question: Push checkm8 iboot configs "upstream" to IPwnDFU

### ISSUE THREAD

**Issue by ghost** [checkra1n/BugTracker#865]
Would it be possible to "push" the  checkm8 "DeviceConfig"s upstream to the IPwnDFU tool, for those not yet supported by the tool (like s8000)?

https://github.com/axi0mX/ipwndfu/blob/master/checkm8.py#L126

If you don't have the time and devices to format/code/test them, maybe just provide them for others to be included and tested?

I'm quite sure that would make some people happy :)

Thanks a lot!

**Comment by Siguza** [comment]
So the unsupported chips are t7000, t7001, s8000, s8003 and s8001.  
The first four use an exploit strategy different to both A7 and A9X and later, which requires to be run in the first entry of DFU - which is something that the python APIs cannot guarantee AFAIK, besides the fact that we'd have to reimplement our exploit in python.  
s8001 we could do I suppose, but then again our set of offsets is substantially different, and on top of that the A9X chip has unusually little SRAM, and I'm not sure whether we could use the ipwndfu code as-is.

So TL;DR: "maybe, but it'd suck".

FWIW though, we still plan to open-source checkra1n eventually. And we _might_ provide options to run payloads at bootrom time earlier than that.

**Comment by ghost** [comment]
Thanks a lot! For the answer and the crazy work you are doing!
